src/main/java/seedu/address/model/client/insurance/InsurancePlan.java

```diff
@@ -50,13 +51,32 @@ public Claim getClaim(String claimId) throws ClaimException {
         throw new ClaimException(CLAIM_NOT_DETECTED_MESSAGE);
     }
 
+    /**
+     * Adds a claim to the insurance plan's list of claims.
+     *
+     * @param claim The claim to be added.
+     */
+    public void addClaim(Claim claim) {
+        claims.add(claim);
+        sortClaims();
+    }
+
     /**
      * Removes a claim from the insurance plan's list of claims.
      *
      * @param claim The claim to be removed.
      */
     public void removeClaim(Claim claim) {
         claims.remove(claim);
+        sortClaims();
+    }
+
+    /**
+     * Sorts the claims of this insurance plan only based on claim status then by claim id.
```

Comment:
Maybe using "client ID" would be better to follow how we write insurance id as "insurance ID" in javadoc

Reply:
that makes sense, good catch!